### ISSUE TITLE
sending funds from v4 wallet to v5 wallet is causing 'tokio-runtime-worker' panic

### ISSUE THREAD

**Issue by marekyggdrasil** [mimblewimble/grin-wallet#558]
**Describe the bug**

During the [grinnode.live Winter 2020 Bug Bash Challenge](https://forum.grin.mw/t/grinnode-live-winter-2020-bug-bash-challenge-and-free-grin-btc-prizes-giveaway/8060) a tester @mojitoo discovered a problem with the wallet. He was able to cause v5 wallet to `'tokio-runtime-worker' panic` by sending funds to it via TOR. His detailed info is provided in [gist](https://gist.github.com/mojitoo/12f8fd547b864a33e36e8798b10aa3d7) and I managed to reproduce the problem [on my own](https://gist.github.com/marekyggdrasil/81a73b431c3a37096b7eb6f94fd70aa8)

**To Reproduce**

1. Prepare two wallets, v4 and v5. Make sure wallet v4 has funds.
2. Set wallet v5 in TOR listener mode.
3. Initiate transaction over TOR from wallet v4 into wallet v5.
4. You should see `'tokio-runtime-worker' panic` in wallet v5 logs.

**Expected behavior**

We would expect the `'tokio-runtime-worker' panic` to not occur.

**Screenshots**

N/A

**Desktop (please complete the following information):**

Tester environment
```
Darwin MacBook-Pro-de-Workstation.local 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
```
Reproduction environment
```
Linux 4.19.0-12-amd64 #1 SMP Debian 4.19.152-1 (2020-10-18) x86_64 unknown unknown GNU/Linux
```

**Additional context**

[Tester report](https://gist.github.com/mojitoo/12f8fd547b864a33e36e8798b10aa3d7) and my report on [reproducing the bug](https://gist.github.com/marekyggdrasil/81a73b431c3a37096b7eb6f94fd70aa8)

**Comment by quentinlesceller** [comment]
Confirmed:
```
20210114 19:41:37.106 ERROR grin_util::logger -
thread 'tokio-runtime-worker' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.': /home/quentin/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/enter.rs:38   0: grin_util::logger::send_panic_to_log::{{closure}}
   1: std::panicking::rust_panic_with_hook
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:597
   2: std::panicking::begin_panic::{{closure}}
   3: std::sys_common::backtrace::__rust_end_short_backtrace
   4: std::panicking::begin_panic
   5: tokio::runtime::enter::enter
   6: tokio::runtime::thread_pool::ThreadPool::block_on
   7: tokio::runtime::context::enter
   8: grin_wallet_impls::client_utils::client::Client::send_request
   9: grin_wallet_impls::client_utils::client::Client::post
  10: <grin_wallet_impls::node_clients::http::HTTPNodeClient as grin_wallet_libwallet::types::NodeClient>::get_version_info
  11: <grin_wallet_api::foreign::Foreign<L,C,K> as grin_wallet_api::foreign_rpc::ForeignRpc>::check_version
  12: <dyn grin_wallet_api::foreign_rpc::ForeignRpc as easy_jsonrpc_mw::Handler>::handle
  13: easy_jsonrpc_mw::handle_call
  14: easy_jsonrpc_mw::handle_parsed_request
  15: easy_jsonrpc_mw::Handler::handle_request
  16: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  17: <hyper::proto::h1::dispatch::Server<S,hyper::body::body::Body> as hyper::proto::h1::dispatch::Dispatch>::poll_msg
  18: hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T>::poll_inner
  19: <hyper::server::conn::upgrades::UpgradeableConnection<I,S,E> as core::future::future::Future>::poll
  20: <hyper::server::conn::spawn_all::NewSvcTask<I,N,S,E,W> as core::future::future::Future>::poll
  21: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  22: tokio::runtime::task::raw::poll
  23: std::thread::local::LocalKey<T>::with
  24: tokio::runtime::thread_pool::worker::Context::run_task
  25: tokio::runtime::thread_pool::worker::Context::run
  26: tokio::macros::scoped_tls::ScopedKey<T>::set
  27: tokio::runtime::thread_pool::worker::run
  28: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
  29: tokio::runtime::task::core::Core<T,S>::poll
  30: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  31: tokio::runtime::task::harness::Harness<T,S>::poll
  32: tokio::runtime::blocking::pool::Inner::run
  33: tokio::runtime::context::enter
  34: std::sys_common::backtrace::__rust_begin_short_backtrace
  35: core::ops::function::FnOnce::call_once{{vtable.shim}}
  36: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/alloc/src/boxed.rs:1307
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/alloc/src/boxed.rs:1307
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys/unix/thread.rs:71
  37: start_thread
  38: clone
```

**Comment by quentinlesceller** [comment]
Okay fixed by #573
